torch/ao/quantization/fx/prepare.py

```diff
@@ -116,22 +117,43 @@
 # list of dtypes to not add observers to
 _DO_NOT_OBS_DTYPE_LIST = [int, float, torch.bool, None]
 
+_OBS_OR_FQ_TYPE = Union[ObserverBase, FakeQuantizeBase, Any]
+# type for annotations for argument or output for node, either a observer/fake_quantize
+# constructor (e.g. observer.with_args(...)), or a tuple of constructors, each one
+# corresponds to an argument for node or a output for node, the tuple can also be nested
+# the requirement is that it matches the structure of input and output
+# e.g. node.meta["target_dtype_info"] = {
+#    "input_obs_or_fq_ctrs": (obs1, obs2),
+#    "output_obs_or_fq_ctrs": (obs3, obs4)
+# }
+_OBS_OR_FQ_ANNO_TYPE = Union[_OBS_OR_FQ_TYPE, Tuple[_OBS_OR_FQ_TYPE, ...]]
```

Comment:
Tuple can contain other tuples, right? SO I would imagine this would have to be
```
_OBS_OR_FQ_ANNO_TYPE = Union[_OBS_OR_FQ_TYPE, Tuple[_OBS_OR_FQ_ANNO_TYPE]]
```

Reply:
yeah, but does it support recursive definition?